beacon-chain/p2p/pubsub.go

```diff
@@ -75,10 +77,24 @@ func (s *Service) SubscribeToTopic(topic string, opts ...pubsub.SubOpt) (*pubsub
 // Content addressable ID function.
 //
 // ETH2 spec defines the message ID as:
-//    message-id: SHA256(message.data)
+//    The `message-id` of a gossipsub message MUST be the following 20 byte value computed from the message data:
+//  If `message.data` has a valid snappy decompression, set `message-id` to the first 20 bytes of the `SHA256` hash of
+//  the concatenation of `MESSAGE_DOMAIN_VALID_SNAPPY` with the snappy decompressed message data,
+//  i.e. `SHA256(MESSAGE_DOMAIN_VALID_SNAPPY + snappy_decompress(message.data))[:20]`.
+//
+//  Otherwise, set `message-id` to the first 20 bytes of the `SHA256` hash of
+//  the concatenation of `MESSAGE_DOMAIN_INVALID_SNAPPY` with the raw message data,
+//  i.e. `SHA256(MESSAGE_DOMAIN_INVALID_SNAPPY + message.data)[:20]`.
 func msgIDFunction(pmsg *pubsub_pb.Message) string {
-	h := hashutil.Hash(pmsg.Data)
-	return string(h[:])
+	decodedData, err := snappy.Decode(nil /*dst*/, pmsg.Data)
+	if err != nil {
```

Comment:
`err != nil` feels a bit generic. On a safer side, do you know what specific errors are ok and are not ok?

Reply:
Afaik, all errors returned from it should signify it is an invalid snappy object.
https://github.com/golang/snappy/blob/master/decode.go#L57 

None of the errors there are 'ok', all point to a corrupted/invalid snappy object as of the spec. If it cant be decoded 
it should be appended with the invalid domain and then hashed.